packages/@react-native-windows/cli/powershell/Eval-MsBuildProperties.ps1

```diff
@@ -0,0 +1,156 @@
+    [Parameter()]
+    [String]$MSBuildPath,
+    [Parameter()]
+    [String]$ExtraMSBuildProps
+)
+
+function Get-MSBuildPath {
+    $vsWhere = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe"
+    if (!(Test-Path $vsWhere)) {
+        throw "Unable to find vswhere.exe."
+    }
+    $vsPath = & $vsWhere -version 16.5 -property installationPath;
+    return "$vsPath\MSBuild\Current\Bin";
+}
+
+function Get-MSBuildProperties {
+    param (
+        [Parameter(Mandatory = $true)]
+        [String]$MSBuildPath,
+        [Parameter(Mandatory = $true)]
+        [String]$SolutionPath,
+        [Parameter(Mandatory = $true)]
+        [String]$ProjectPath,
+        [Parameter()]
+        [String[]]$PropertyNames = @(),
+        [Parameter()]
+        [String]$ExtraMSBuildProps
+    )
+
+    if (!(Test-Path (Join-Path $MSBuildPath "MSBuild.exe"))) {
+        throw "Unable to find MSBuild.exe in $MSBuildPath"
+    }
+
+    if (!(Test-Path (Join-Path $MSBuildPath "MSBuild.exe"))) {
+        throw "Unable to find Microsoft.Build.dll in $MSBuildPath"
+    }
+
+    # Method to intercept resolution of assemblies to add MSBuild's path
+    $onAssemblyResolveEventHandler = [System.ResolveEventHandler] {
+        param($s, $e)
+
+        # Figure out which assembly to look for
+        $assemblyName = $e.Name.Substring(0, $e.Name.IndexOf(", "));
+        $assemblyPath = "$MSBuildPath\$assemblyName.dll"
+
+        # Search for the assembly in the MSBuild directory
+        if (Test-Path $assemblyPath) {
+            # Found the assembly!
+            return [System.Reflection.Assembly]::LoadFrom("$MSBuildPath\$assemblyName.dll");
+        }
+
+        # Unable to find the assembly
+        return $null
+    }
+
+    # Wire-up assembly resolution event handler
+    [System.AppDomain]::CurrentDomain.add_AssemblyResolve($onAssemblyResolveEventHandler)
+
+    # Load Microsoft.Build.dll into script
+    Add-Type -Path "$MSBuildPath\Microsoft.Build.dll" | Out-Null
+
+    # Build a local project collection
+    $projectCollection = [Microsoft.Build.Evaluation.ProjectCollection]::new()
+
+    try {
+        # Build a temporary "metaproj" of the solution file so it can be processed
+        ${env:MSBUILDEMITSOLUTION} = 1
+        & $MSBuildPath\MSBuild.exe $SolutionPath | Out-Null
+
+        # Process solution
+        $solution = [Microsoft.Build.Evaluation.Project]::new("$SolutionPath.metaproj", $null, "Current", $projectCollection)
+    }
+    finally {
+        # Clean up "metaproj" files
+        ${env:MSBUILDEMITSOLUTION} = 0
+        Remove-Item -Path @("$SolutionPath.metaproj", "$SolutionPath.metaproj.tmp") | Out-Null
+        Get-ChildItem * -Include *.metaproj -Recurse | Remove-Item | Out-Null
+    }
+
+    # Evaluate all of the Solution* properties and save into a collection
+    $globalProps = New-Object 'System.Collections.Generic.Dictionary[String,String]'
+    $solution.Properties | ForEach-Object -Process {
```

Comment:
nit, you can use `%` instead of Foreach-object -process

Reply:
I'd vote against aliases for properly-formed scripts.
Those less experienced with PowerShell may find the explicit cmdlets easier to read.